deploy-cockroachdb-on-aws.md

```diff
@@ -71,23 +89,96 @@ To connect your application to CockroachDB, use a [PostgreSQL wire protocol driv
 - Running at least 3 nodes to ensure survivability.
 - Selecting the same continent for all of your instances for best performance.
 
-## Step 3. Set up the First Node
+## Step 3. Generate Your Certificates
+
+Locally, you'll need to [create the following certificates and keys](create-security-certificates.html):
+
+- A certificate authority (CA) key pair (`ca.cert` and `ca.key`)
+- A client key pair for the `root` user
+- A node key pair for each node, issued to its IP addresses and any common names the machine uses
+
+{{site.data.alerts.callout_success}}Before beginning, it's useful to collect each of your machine's internal and external IP addresses, as well as any server names you want to issue certificates for.{{site.data.alerts.end}}
+
+1. Create a `certs` directory:
+
+   ~~~ shell
+   $ mkdir certs
+   ~~~
+
+2. Create the CA key pair:
+
+   ~~~ shell
+   $ cockroach cert create-ca \
+   --ca-cert=certs/ca.cert \
+   --ca-key=certs/ca.key
+   ~~~
+
+4. Create a client key pair for the `root` user:
+
+   ~~~ shell
+   $ cockroach cert create-client \
+   root \
+   --ca-cert=certs/ca.cert \
+   --ca-key=certs/ca.key \
+   --cert=certs/root.cert \
+   --key=certs/root.key
+   ~~~
+
+3. Create a node key pair for each node issued for all common names you might use to refer to the node, including:
+
+   - `<node internal IP address>` which is the instance's **Internal IP**.
+   - `<node external IP address>` which is the instance's **External IP address**.
+   - `<node hostname>` which is the instance's hostname. <br/><br/>You can find this by SSHing into a server and running `hostname`. For many AWS EC2 servers, this is `ip-` followed by the internal IP address delimited by dashes; e.g., `ip-172-31-18-168`.
+   - `<other common names for node>` which include any domain names you point to the instance.
+   - `localhost` and `127.0.0.1`
+
+   ~~~ shell
+   $ cockroach cert create-node \
+   <node internal IP address> \
+   <node external IP address> \
+   <node hostname>  \
+   <other common names for node> \
+   localhost \
+   127.0.0.1 \
+   --ca-cert=certs/ca.cert \
+   --ca-key=certs/ca.key \
+   --cert=certs/<node name>.cert \
+   --key=certs/<node name>.key
+   ~~~
+
+4. Upload the certificates to each node:
+
+   ~~~ shell
+   # Create the certs directory:
+   $ ssh -i <path to AWS .pem> <username>@<node external IP address> "mkdir certs"
+
+   # Upload the CA certificate, client (root) certificate and key, and node certificate and key:
```

Comment:
Nodes don't need (and shouldn't have) a client cert/key, just the node cert/key and CA cert. 

Call out that the `ca.key` should *not* be uploaded to every node, only `ca.cert`. 

After uploading the node keys, you should delete them from the machine where you generated them, so that each node key/cert pair exists only on that node.